File: e2e-test/multi-module-setup/ios/modules/LynxjsLinkingModule.swift
import Foundation

@objcMembers
public final class LynxjsLinkingModule: NSObject, LynxModule {

    public static var name: String {
        return "LynxjsLinkingModule"
    }

    public static var methodLookup: [String : String] {
        return [
        "openURL": NSStringFromSelector(#selector(openURL(url:callback:))),
        "openSettings": NSStringFromSelector(#selector(openSettings(callback:))),
        "sendIntent": NSStringFromSelector(#selector(sendIntent(action:extras:callback:))),
        "share": NSStringFromSelector(#selector(share(content:options:callback:)))
        ]
    }

    func openURL(url: String, callback: (Any) -> Void) {
        fatalError("Not implemented")
    }

    func openSettings(callback: (Any) -> Void) {
        fatalError("Not implemented")
    }

    func sendIntent(action: String, extras: NSArray?, callback: (Any) -> Void?) {
        fatalError("Not implemented")
    }

    func share(content: String, options: NSDictionary?, callback: (Any) -> Void?) {
        fatalError("Not implemented")
    }
}

File: e2e-test/multi-module-setup/android/src/main/kotlin/com/himanshu/tools/LynxjsLinkingModule.kt
package com.himanshu.tools

import android.content.Context
import com.lynx.jsbridge.LynxMethod
import com.lynx.jsbridge.LynxModule
import com.lynx.tasm.behavior.LynxContext
import com.lynx.react.bridge.Callback
import com.lynx.react.bridge.ReadableArray
import com.lynx.react.bridge.ReadableMap

class LynxjsLinkingModule(context: Context) : LynxModule(context) {
  private fun getContext(): Context {
    val lynxContext = mContext as LynxContext
    return lynxContext.getContext()
  }

  @LynxMethod
  fun openURL(url: String, callback: Callback) {
    TODO()
  }

  @LynxMethod
  fun openSettings(callback: Callback) {
    TODO()
  }

  @LynxMethod
  fun sendIntent(action: String, extras: ReadableArray?, callback: Callback?) {
    TODO()
  }

  @LynxMethod
  fun share(content: String, options: ReadableMap?, callback: Callback?) {
    TODO()
  }
}